some hosts are more anal about ar usage than others
http://bugs.gentoo.org/107428

respect user LDFLAGS
http://bugs.gentoo.org/126872

--- lib/makefile.src
+++ lib/makefile.src
@@ -181,8 +181,7 @@
 # Rebuild the library from all the .o files.
 buildlib: $(OBJS)
 	-rm -f $(LIBNAME)
-	ar rc $(LIBNAME)
-	ar q $(LIBNAME) $(OBJS)
+	ar rc $(LIBNAME) $(OBJS)
 #ifndef NO_RANLIB
 	ranlib $(LIBNAME)
 #endif
--- tools/clock/makefile.src
+++ tools/clock/makefile.src
@@ -6,7 +6,7 @@ install: rleClock
 	mv rleClock ../rleClock.out
 
 rleClock:rleClock.o font.o
-	${CC} ${CFLAGS} rleClock.o font.o -lm ${LIBS} -o rleClock
+	${CC} ${CFLAGS} ${LDFLAGS} rleClock.o font.o ${LIBS} -o rleClock -lm
 
 font.c:font.src makeFont
 	chmod +x makeFont
--- tools/makefile.src
+++ tools/makefile.src
@@ -62,21 +62,21 @@ applymap.out rlebg.out: $(RI)/rle_raw.h
 pyrlib.o: pyrlib.c $(RI)/pyramid.h $(RI)/rle.h $(RI)/rle_config.h
 	$(CC) $(CFLAGS) pyrlib.c -c
 pyrmask.out: pyrlib.o pyrmask.c $(RI)/pyramid.h
-	$(CC) $(CFLAGS) -I$(RI) pyrmask.c pyrlib.o $(LIBS) -lm -o pyrmask.new
+	$(CC) $(LDFLAGS) $(CFLAGS) -I$(RI) pyrmask.c pyrlib.o $(LIBS) -lm -o pyrmask.new
 	mv pyrmask.new pyrmask.out
 
 fant.out: fant.o mallocNd.o
-	$(CC) $(CFLAGS) -I$(RI) fant.o mallocNd.o $(LIBS) -lm -o fant.new
+	$(CC) $(LDFLAGS) $(CFLAGS) -I$(RI) fant.o mallocNd.o $(LIBS) -lm -o fant.new
 	mv fant.new fant.out
 	
 # rlebox and crop use some common code.
 rle_box.o: $(RI)/rle.h $(RI)/rle_config.h $(RI)/rle_raw.h
 
 crop.out: crop.c rle_box.o
-	${CC} ${CFLAGS} crop.c rle_box.o ${LIBS} -o crop.new
+	${CC} ${LDFLAGS} ${CFLAGS} crop.c rle_box.o ${LIBS} -o crop.new
 	mv crop.new crop.out
 rlebox.out: rlebox.c rle_box.o
-	${CC} ${CFLAGS} rlebox.c rle_box.o ${LIBS} -o rlebox.new
+	${CC} ${LDFLAGS} ${CFLAGS} rlebox.c rle_box.o ${LIBS} -o rlebox.new
 	mv rlebox.new rlebox.out
 
 # rleClock has it's own directory, must be built special
@@ -100,7 +100,7 @@ clean:	clean-pgm
 .SUFFIXES:
 .SUFFIXES: .out .c .o
 .c.out:
-	$(CC) $(CFLAGS) $< $(LIBS) -lm -o $*.new
+	$(CC) $(LDFLAGS) $(CFLAGS) $< $(LIBS) -lm -o $*.new
 	mv $*.new $@
 
 .c.o:
--- cnv/makefile.src
+++ cnv/makefile.src
@@ -76,13 +76,13 @@ PBMDIR = 
 # ppmtorle - ppm format to RLE
 # rletoppm - RLE to ppm format
 pgmtorle.out: pgmtorle.c
-	$(CC) $(CFLAGS) $(INCPBMPLUS) $*.c $(LIBS) $(LIBPBMPLUS) -o $*.new
+	$(CC) $(CFLAGS) $(LDFLAGS) $(INCPBMPLUS) $*.c $(LIBS) $(LIBPBMPLUS) -o $*.new
 	mv $*.new $@
 ppmtorle.out: ppmtorle.c
-	$(CC) $(CFLAGS) $(INCPBMPLUS) $*.c $(LIBS) $(LIBPBMPLUS) -o $*.new
+	$(CC) $(CFLAGS) $(LDFLAGS) $(INCPBMPLUS) $*.c $(LIBS) $(LIBPBMPLUS) -o $*.new
 	mv $*.new $@
 rletoppm.out: rletoppm.c
-	$(CC) $(CFLAGS) $(INCPBMPLUS) $*.c $(LIBS) $(LIBPBMPLUS) -o $*.new
+	$(CC) $(CFLAGS) $(LDFLAGS) $(INCPBMPLUS) $*.c $(LIBS) $(LIBPBMPLUS) -o $*.new
 	mv $*.new $@
 #endif
 
@@ -95,10 +95,10 @@ rletoppm.out: rletoppm.c
 # iristorle/rletoiris - Convert between RLE and SGI image format.
 #
 iristorle.out: iristorle.c
-	$(CC) $(CFLAGS) -I/usr/include/gl $*.c $(LIBS) -limage -o $*.new
+	$(CC) $(CFLAGS) $(LDFLAGS) -I/usr/include/gl $*.c $(LIBS) -limage -o $*.new
 	mv $*.new $@
 rletoiris.out: rletoiris.c
-	$(CC) $(CFLAGS) -I/usr/include/gl $*.c $(LIBS) -limage -o $*.new
+	$(CC) $(CFLAGS) $(LDFLAGS) -I/usr/include/gl $*.c $(LIBS) -limage -o $*.new
 	mv $*.new $@
 #endif
 
@@ -108,10 +108,10 @@ TIFFDIR =
 # tifftorle - Convert TIFF images to RLE
 # rletotiff - Convert RLE images to TIFF
 rletotiff.out: rletotiff.c
-	$(CC) $(CFLAGS) $(INCTIFF) $*.c $(LIBS) $(LIBTIFF) -lm -o $*.new
+	$(CC) $(CFLAGS) $(LDFLAGS) $(INCTIFF) $*.c $(LIBS) $(LIBTIFF) -lm -o $*.new
 	mv $*.new $@
 tifftorle.out: tifftorle.c
-	$(CC) $(CFLAGS) $(INCTIFF) $*.c $(LIBS) $(LIBTIFF) -lm -o $*.new
+	$(CC) $(CFLAGS) $(LDFLAGS) $(INCTIFF) $*.c $(LIBS) $(LIBTIFF) -lm -o $*.new
 	mv $*.new $@
 #endif
 
@@ -125,7 +125,7 @@ tifftorle.out: tifftorle.c
 # Will build with the default rule.
 # rletorla - RLE to Wavefront RLA
 rletorla.out: rletorla.c
-	$(CC) $(CFLAGS) $*.c $(LIBS) $(LIBWAVEFRONT) -lm -o $*.new
+	$(CC) $(CFLAGS) $(LDFLAGS) $*.c $(LIBS) $(LIBWAVEFRONT) -lm -o $*.new
 	mv $*.new $@
 #endif WAVEFRONT
 
@@ -144,7 +144,7 @@ pristine: pristine-pgm
 .SUFFIXES:
 .SUFFIXES: .out .c
 .c.out:
-	$(CC) $(CFLAGS) $*.c $(LIBS) -lm -o $*.new
+	$(CC) $(CFLAGS) $(LDFLAGS) $*.c $(LIBS) -lm -o $*.new
 	mv $*.new $@
 
 # Dependency lines.  Make sure to #ifdef them.
--- cnv/rletoabA62/makefile.src
+++ cnv/rletoabA62/makefile.src
@@ -15,7 +15,7 @@ all		:	$(PGMS)
 # Executables.  The .out will be stripped off in the install action.
 
 rletoabA62.out	:	rletoabA62.o rle.o
-			$(CC) $(CFLAGS) -o rletoabA62.new \
+			$(CC) $(CFLAGS) $(LDFLAGS) -o rletoabA62.new \
 			  rletoabA62.o rle.o $(LIBS)
 			mv rletoabA62.new rletoabA62.out
 
--- cnv/rletogif/makefile.src
+++ cnv/rletogif/makefile.src
@@ -15,7 +15,7 @@ all:	$(PGMS)
 # The executable.  The ".out" will be stripped off in the install action.
 
 rletogif.out: ${OBJ}
-	${CC} ${CFLAGS} ${OBJ} ${LIBS} -o rletogif.new
+	${CC} ${CFLAGS} ${LDFLAGS} ${OBJ} ${LIBS} -o rletogif.new
 	mv rletogif.new rletogif.out
 
 # Incremental install, copies executable to DEST dir.
